### ISSUE TITLE
How to flush the route table

### ISSUE THREAD

**Issue by jsyadav** [vishvananda/netlink#331]
Is there a way to flush the route table?

equivalent to
"iproute2 route flush table x"

**Comment by jjastrze-ovh** [comment]
RouteList + RouteDel ?

**Comment by xaviershay** [comment]
I don't think the above method works as is for `ip {,-6} route flush cache`.

iproute2 uses `proc` directly for ipv4. Relevant [source code](https://github.com/shemminger/iproute2/blob/707f612c00a91fe3c17d7af5319e423545a496c5/ip/iproute.c#L1272) and strace excerpt:

```
30362 openat(AT_FDCWD, "/proc/sys/net/ipv4/route/flush", O_WRONLY) = 4
30362 write(4, "-1", 2)                 = 2
30362 close(4)                          = 0
```

For ipv6 flushing, I'm not sure it's possible with this library with current API. From reading the source iproute2 source code above, it looks like any cloned route is considered cached. `RouteListFiltered` is our only access to `RTM_GETROUTE`, and it [filters out cloned routes](https://sourcegraph.com/github.com/vishvananda/netlink/-/blob/route_linux.go#L725). I don't understand this cache/cloned thing sufficiently to propose an appropriate API, nor to have confidence in my above assessment. Would love someone with more relevant experience to chime in.

**Comment by xaviershay** [comment]
`/proc/sys/net/ipv6/route/flush` exists and people appear to be using it for this purpose, though I can't seem to find it actually documented anywhere. Perhaps the right approach here is to not use netlink at all for flushing route cache?

**Comment by oneumyvakin** [comment]
Should be AddrList / AddrDel used for `ip addr flush dev eth0` as well?

**Comment by xaviershay** [comment]
Yeah probably, sorry not able to confirm at the moment. Check the link in my first comment, you can strace it to confirm.
